src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -862,7 +867,7 @@ public function resolve(array $options = []): array
         $clone = clone $this;
 
         // Make sure that no unknown options are passed
-        $diff = array_diff_key($options, $clone->defined);
+        $diff = true === $this->ignoreUndefined ? [] : array_diff_key($options, $clone->defined);
```

Comment:
no need to compare to true/false anywhere

Reply:
done